docs/mp/oci/03_vault.adoc

```diff
@@ -0,0 +1,235 @@
+* Compartment OCID - OCI-specific compartment
+* Encryption Key OCID - required when doing encryption/decryption
+* Signature Key OCID - required when doing signatures/verification
+* Cryptographic endpoint - required for all except secrets
+
+First specify OCIDs and URLs of Vault items in `microprofile-config.properties`:
+
+[source,properties]
+----
+oci.vault.vault-ocid: "<...>"
+oci.vault.compartment-ocid: "<...>"
+oci.vault.encryption-key-ocid: "<...>"
+oci.vault.signature-key-ocid: "<...>"
+oci.vault.cryptographic-endpoint: "<...>"
+----
+
+The OCIDs can be set up and found in OCI under Security tab.
+
+image::oci/ocivault.png[OCI Vault, align="center"]
+
+REST endpoint should be setup as follows:
+
+[source,java]
+----
+@Path("/vault")
+public class VaultResource {
+    private final OciVault vault;
+    private final String vaultOcid;
+    private final String compartmentOcid;
+    private final String encryptionKeyOcid;
+    private final String signatureKeyOcid;
+
+    @Inject
+    VaultResource(@Named("custom") OciVault vault, <1>
+                  @ConfigProperty(name = "app.vault.vault-ocid") <2>
+                          String vaultOcid,
+                  @ConfigProperty(name = "app.vault.compartment-ocid") <3>
+                          String compartmentOcid,
+                  @ConfigProperty(name = "app.vault.encryption-key-ocid") <4>
+                          String encryptionKeyOcid,
+                  @ConfigProperty(name = "app.vault.signature-key-ocid") <5>
+                          String signatureKeyOcid) {
+        this.vault = vault;
+        this.vaultOcid = vaultOcid;
+        this.compartmentOcid = compartmentOcid;
+        this.encryptionKeyOcid = encryptionKeyOcid;
+        this.signatureKeyOcid = signatureKeyOcid;
+    }
+}
+----
+
+<1> `OciVault` support class is configured and injected automatically
+<2> <3> <4> <5> Properties a read from the configuration
+
+== OCI Vault usage
+
+=== Encryption
+To encrypt a text, submitted with a GET request to `/encrypt` endpoint:
+
+[source,java]
+----
+@GET
+@Path("/encrypt/{text}")
+public String encrypt(@PathParam("text") String secret) {
+    return vault.encrypt(Encrypt.Request.builder()
+                                 .keyId(encryptionKeyOcid)
+                                 .data(Base64Value.create(secret)))
+            .cipherText();
+}
+----
+
+=== Decryption
+To decrypt a text, submitted with a GET request to `/decrypt` endpoint:
+
+[source,java]
+----
+@GET
+@Path("/decrypt/{text: .*}")
+public String decrypt(@PathParam("text") String cipherText) {
+    return vault.decrypt(Decrypt.Request.builder()
+                                 .keyId(encryptionKeyOcid)
+                                 .cipherText(cipherText))
+            .decrypted()
+            .toDecodedString();
+}
+----
+
+=== Signature
+To retrieve a signature for submitted text via GET request to `/sign` endpoint:
```

Comment:
To retrieve a signature for a submitted text, submit a GET request to the `/sign` endpoint: